spec/requests/rails_spec.rb

```diff
@@ -13,9 +13,21 @@
 
 RSpec.describe 'Tables', type: :request do
   describe '#index' do
-    it 'returns a list of tables' do
-      get '/tables', params: { page: '1', per: '10' }, headers: { authorization: 'k0kubun' }
-      expect(response.status).to eq(200)
+    context it 'returns a list of tables' do
+      it 'with flat query parameters' do
+        get '/tables', params: { page: '1', per: '10' }, headers: { authorization: 'k0kubun' }
+        expect(response.status).to eq(200)
+      end
+
+      it 'with deep query parameters' do
+        get '/tables', params: { filter: { "name" => "Example Table" } }, headers: { authorization: 'k0kubun' }
```

Comment:
It's just a guess, but maybe the test is passing ignoring this because the above test generates parameters, and running this probably doesn't override it. So copying the same endpoint to somewhere else and calling it from this spec would probably approach the test issue.

Reply:
True enough, though I'd like to be able to generate a spec like

```yaml
openapi: 3.0.3
info:
  title: rspec-openapi
  version: 1.0.0
paths:
  "/tables":
    get:
      summary: index
      tags:
      - Table
      parameters:
      - name: page
        in: query
        schema:
          type: integer
        example: 1
      - name: per
        in: query
        schema:
          type: integer
        example: 10
      - name: "filter[name]"
        in: query
        schema:
          type: string
        example: Example Table
      - name: "filter[price]"
        in: query
        schema:
          type: integer
        example: 0
      responses:
        '200':
          description: returns a list of tables
          content:
            application/json:
              schema:
                oneOf:
                  - type: array
                  items:
                    type: object
                    # etc
```

it's been a while since I studied the oas spec, so perhaps I'm misunderstanding it...

https://swagger.io/docs/specification/describing-responses/

```
Can I have different responses based on a request parameter? Such as:
GET /something -> {200, schema_1}
GET /something?foo=bar -> {200, schema_2}

In OpenAPI 3.0, you can use `oneOf` to specify alternate schemas for the response and document possible dependencies verbally in the response description. However, there is no way to link specific schemas to certain parameter combinations.
```

Reply:
(If I remove the 'with flat query parameters' spec and rm spec/rails/doc/openapi.yaml and run `OPENAPI=1 bundle exec rspec spec/requests/rails_spec.rb` it generates a new spec but still with only one param. )

Reply:
I agree. That's something we should fix as well. But if you're not trying to address that, maybe we should separate that for now and possibly merge them in a PR that address the problem.